presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -553,6 +562,15 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
             return Optional.of(mostSpecificFunction.getBoundSignature());
         }
 
+        // Prefer functions where all arguments are of the same type
```

Comment:
It looks like a hack, but i can't come up with any reasonable example where this would produce weird behavior. So, let's just leave it as is.